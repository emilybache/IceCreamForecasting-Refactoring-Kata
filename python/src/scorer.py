import enum
import random


class IceCream(enum.Enum):
    Strawberry = 1
    Chocolate = 2
    Vanilla = 3


flavour = None


def get_score():
    sunny_today = lookup_weather()
    global flavour
    return get_score_for_flavour_and_weather(flavour, sunny_today)


def get_score_for_flavour_and_weather(current_flavour, sunny_today):
    if current_flavour == IceCream.Strawberry:
        if sunny_today:
            return 10
        else:
            return 5
    elif current_flavour == IceCream.Chocolate:
        return 6
    elif current_flavour == IceCream.Vanilla:
        if sunny_today:
            return 7
        else:
            return 5
    else:
        return -1


def lookup_weather():
    # placeholder implementation - real version would make API call to weather service
    sunny = random.choice([True, False])
    return bool(sunny)


def update_selection():
    score = get_score()
    if score > 5:
        global flavour
        # placeholder implementation - real version would use machine learning to predict sales
        flavour = random.choice([IceCream.Strawberry, IceCream.Chocolate, IceCream.Vanilla])


def get_sales_forecast():
    forecasts = {
        IceCream.Strawberry: 9,
        IceCream.Vanilla: 11,
        IceCream.Chocolate: 3
    }
    return forecasts[flavour]
