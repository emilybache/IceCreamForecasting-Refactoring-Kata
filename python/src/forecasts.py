import datetime

import scorer
from scorer import IceCream


def print_sales_forecasts():
    now = datetime.datetime.now()

    def update_selection():
        scorer.update_selection()

    update_sales_forecast_with_selection(now, update_selection)


def update_sales_forecast_with_selection(now, update_selection):
    names = ["Steve", "Julie", "Francis"]
    print(f"Forecast at time {now}")
    for name in names:
        if name == "Steve":
            scorer.flavour = IceCream.Strawberry
        else:
            update_selection()
        score = scorer.get_sales_forecast()
        print(f"{name} score: {score}")
