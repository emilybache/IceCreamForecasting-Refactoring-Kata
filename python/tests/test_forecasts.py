import datetime

import approvaltests

import scorer
from forecasts import update_sales_forecast_with_selection


def stub_update_selection():
    scorer.flavour = scorer.IceCream.Chocolate


def test_scoreboard(capsys):
    update_sales_forecast_with_selection(datetime.datetime.fromisoformat("2023-05-08 10:12:37.023834"),
                                         update_selection=stub_update_selection)
    captured = capsys.readouterr()
    approvaltests.verify(captured.out)
