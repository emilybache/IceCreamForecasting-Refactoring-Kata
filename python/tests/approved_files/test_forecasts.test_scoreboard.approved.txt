Forecast at time 2023-05-08 10:12:37.023834
Steve score: 9
Julie score: 3
Francis score: 3
